python/lsst/meas/deblender/plugins.py

```diff
@@ -166,111 +167,122 @@ def buildMultibandTemplates(debResult, log, useWeights=False, usePsf=False,
 
     Returns
     -------
     modified: `bool`
         If any templates have been created then ``modified`` is ``True``,
         otherwise it is ``False`` (meaning all of the peaks were skipped).
     """
-    import deblender
+    import scarlet
 
     # Extract coordinates from each MultiColorPeak
     bbox = debResult.footprint.getBBox()
     peakSchema = debResult.footprint.peaks.getSchema()
     xmin = bbox.getMinX()
     ymin = bbox.getMinY()
-    peaks = [[pk.x-xmin, pk.y-ymin] for pk in debResult.peaks]
+    peaks = [[pk.y-ymin, pk.x-xmin] for pk in debResult.peaks]
 
     # Create the data array from the masked images
     maskedImages = [mimg.Factory(mimg, debResult.footprint.getBBox(), PARENT)
                     for mimg in debResult.maskedImages]
     data = np.array([mimg.image.array for mimg in maskedImages])
-    data = deblender.nmf.reshape_img(data, truncate=truncate)
-
-    defaultKwargs = {
-        "components": None,
-        "constraints": "S",
-    }
-    
-    for k,v in defaultKwargs.items():
-        if k not in multibandKwargs.keys():
-            multibandKwargs[k] = v
-
-    # Update optional parameters
-    if "weights" not in multibandKwargs:
-        if useWeights:
-            # Use the inverse variance as the weights
-            weights = 1/np.array([mimg.variance.array for mimg in maskedImages])
-
-            # Set bad pixeks weights to zero
-            maskPlane = maskedImages[0].getMask().getMaskPlaneDict()
-            badPixels = (1<<maskPlane["BAD"] |
-                         1<<maskPlane["CR"] |
-                         1<<maskPlane["NO_DATA"] |
-                         1<<maskPlane["SAT"] |
-                         1<<maskPlane["SUSPECT"])
-            fpMask = afwImage.Mask(bbox)
-            debResult.footprint.spans.setMask(fpMask, 1)
-            fpMask = ~fpMask.getArray().astype(bool)
-            mask = np.zeros(weights.shape, np.int64)
-            for fidx, mimg in enumerate(maskedImages):
-                mask[fidx] = mimg.getMask().array
-            mask = ((badPixels & mask) | fpMask).astype(bool)
-            weights[mask] = 0
-
-            # Make sure that the weights and data have the same size
-            weights = deblender.nmf.reshape_img(weights, new_shape=data.shape, truncate=truncate)
-            multibandKwargs["weights"] = weights
-    if "psfs" not in multibandKwargs and usePsf:
+
+    # Use the inverse variance as the weights
+    if useWeights:
+        weights = 1/np.array([mimg.variance.array for mimg in maskedImages])
+    else:
+        weights = weights = np.ones_like(data)
+
+    # Use the mask plane to mask bad pixels and
+    # the footprint to mask out pixels outside the footprint
+    maskPlane = maskedImages[0].getMask().getMaskPlaneDict()
+    badPixels = (1<<maskPlane["BAD"] |
+                 1<<maskPlane["CR"] |
+                 1<<maskPlane["NO_DATA"] |
+                 1<<maskPlane["SAT"] |
+                 1<<maskPlane["SUSPECT"])
```

Comment:
The user may want to specify the mask planes to flag.

Reply:
Oh, and are you hoping the mask is the same for all masked images here?

Reply:
The mask doesn't have to be the same in all bands for this to work, but the mask plane does. I do like your idea to let the user specify the keys to use for bad pixels in the mask plane and I think it's safe to assume that each image will have the same mask plane, that will be implemented on the next commit. @TallJimbo , please shout if you think there is a chance that a set of coadds might have different mask planes when passed to the deblender and I'll update the code.

Reply:
You should write this as:

    badPixels = maskedImages[0].mask.getPlaneBitMask(self.config.badMask)

Reply:
I *think* it's safe to assume that coadds from different bands will have the same mask planes, but adding an `assert` to at least check that they have the same number of mask planes would probably be wise.

Reply:
Ok, just to be safe I calculate the mask for each maskedImage independently:
```
    if badMask is None:
        badMask = ["BAD", "CR", "NO_DATA", "SAT", "SUSPECT"]
    fpMask = afwImage.Mask(bbox)
    debResult.footprint.spans.setMask(fpMask, 1)
    fpMask = ~fpMask.getArray().astype(bool)
    mask = np.zeros(weights.shape, dtype=bool)
    for fidx, mimg in enumerate(maskedImages):
        badPixels = mimg.mask.getPlaneBitMask(badMask)
        mask[fidx] = (mimg.getMask().array & badPixels) | fpMask
    weights[mask] = 0
```